homotopy-web/src/model/proof.rs

```diff
@@ -560,7 +612,11 @@ impl ProofState {
             let diagram: DiagramN = workspace.diagram.clone().try_into().unwrap();
             let generator: DiagramN = self
                 .signature
-                .get(&option.generator)
+                .iter()
+                .find_map(|item| match item {
+                    SignatureItem::Item((g, i)) if g == &option.generator => Some(i),
+                    _ => None,
```

Comment:
I think maybe we should inline the `Node` in the generator data so that we don't need to do this iteration everywhere.

Reply:
I think this would make the serialisation non-deterministic, and make it depend on the order in which a workspace is created. Ideally we'd like to keep it deterministic.